### PR TITLE
Messaging

### PR DESCRIPTION
**Pull request by Desar14**

### CODE REVIEW

File: ProductUpdateModel.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    public class ProductUpdateModel
+    {
+        public int Id { get; set; }
+        public string? Name { get; set; }
+        public ItemImageModel? Image { get; set; }
+        public decimal Price { get; set; }
+        public int Quantity { get; set; }
+    }
+
+    public class ItemImageModel
```
Comment (KirylPrykhodzka) [inline ProductUpdateModel.cs]: one class per file please

File: Program.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+using System.Diagnostics;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+
+namespace LearnNet_MessageProcessor
+{
+    internal class Program
+    {
+        static string CartingServiceBaseUrl = "https://localhost:7268/api/v1/CartItems";
```
Comment (KirylPrykhodzka) [inline Program.cs]: make use of appsettings

File: Program.cs [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+
+        static async Task Main(string[] args)
+        {
+            ServiceBusClient client = new ServiceBusClient(
+                    "learningnet.servicebus.windows.net",
```
Comment (KirylPrykhodzka) [inline Program.cs]: make use of appsettings

File: Program.cs [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+                await processor.StopProcessingAsync();
+                Console.WriteLine("Stopped receiving messages");
+            }
+            finally
+            {
+                await processor.DisposeAsync();
+                await client.DisposeAsync();
+            }
+
+            async Task MessageHandler(ProcessMessageEventArgs args)
```
Comment (KirylPrykhodzka) [inline Program.cs:53]: please specify access scope explicitly (add private keyword)

File: Program.cs [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+                }
+
+                if(response.StatusCode >= System.Net.HttpStatusCode.InternalServerError)
+                {
+                    Console.WriteLine("Can't be processed 500");
+                    //left for later processing, maybe it will get alive
+                    return;
+                }
+
+                if (response.StatusCode >= System.Net.HttpStatusCode.BadRequest)
```
Comment (KirylPrykhodzka) [inline Program.cs]: Does not make too much sense to me to handle 5xx and 4xx status codes differently.
Let's discuss it at our meeting

**Review by KirylPrykhodzka — CHANGES_REQUESTED** [review]

File: Program.cs (hunk 5 above)
Comment (Desar14) [reply Program.cs]: make same handling

**Review by KirylPrykhodzka — APPROVED** [review]

**Pull request merged**